### ISSUE TITLE
Support `<noscript amp-noscript >` tags

### ISSUE THREAD

**Issue by anrghg** [Automattic/amp-wp#7227]
### Feature Description

Due to AMP-WP using `<noscript>`, issues arise that need to be worked around, e.g. by conditionally canceling `noscript` rules, depending on a dedicated class (in this example: `anrghg-is-amp`).

Example:
```php
if ( $l_b_amp_active ) {
	$l_s_prepended .= ' on="tap:anrghg_toggle_contents_' . $l_i_post_id;
	$l_s_prepended .= '.toggleChecked(force=true),';
	$l_s_prepended .= ' anrghg_contents_list_' . $l_i_post_id;
	$l_s_prepended .= '.toggleClass(class=anrghg-instant, force=true),';
	$l_s_prepended .= ' anrghg_contents_list_' . $l_i_post_id;
	$l_s_prepended .= '.toggleClass(class=anrghg-is-amp, force=true)"';
} else {
	$l_s_prepended .= " onclick=\"anrghgExpandContents$l_i_post_id()\"";
}
```

To ease implementations, `<noscript>` tags with the `amp-noscript` argument could be skipped when unwrapping `<noscript>`. So when we have things that should be used only when JavaScript is actually deactivated, we would wrap them into `<noscript amp-noscript>` code here `</noscript>`.

In the example, when JavaScript is off, the target item should display, whereas when the AMP-HTML framework is available, the target should not display because there is a way to expand and collapse the whole table.

```css
.anrghg-display-toggle:not(:checked) + .anrghg-tocontents .anrghg-contents-list .anrghg-contents-heading:target {
	opacity: 1;
	margin-bottom: $l_s_row_spacing;
	line-height: $l_f_item_line_height;
	visibility: visible;
}

.anrghg-display-toggle:not(:checked) + .anrghg-tocontents .anrghg-contents-list.anrghg-is-amp .anrghg-contents-heading:target {
	opacity: 0;
	padding-top: 0;
	margin-top: 0;
	margin-bottom: 0;
	line-height: 0;
	visibility: hidden;
}
```

All this would be easier if `<noscript>` was available at two levels, the traditional one to be unwrapped, the AMP-specific one to be left as-is for when JS is actually off.

Because restricting JS to the AMP-HTML framework is not really turning it off, those using the framework should still be able to address the case when JS is turned  off.

**Comment by westonruter** [comment]
Thanks for the suggestion. This was actually already implemented via #6528 fixing #6030. The attribute name is `data-amp-no-unwrap`.

**Comment by anrghg** [comment]
@westonruter You are welcome! Thank you for coming up with #6030 and implementing the `data-amp-no-unwrap` attribute! (In the cited use case coded while you were on it, the fix will avoid that the target entry shows up below the label of a collapsed TOC after reloading an AMP page no matter whether JS is on or off, which looked weird and is addressed in the 2ⁿᵈ paragraph of FAQ item https://wordpress.org/plugins/anrghg/#what%20is%20the%20lone%20heading%20after%20the%20table%20of%20contents%20label%3F because it should happen only when JS is fully turned off.) — I’ll hurry to add the new attribute fixing noscript. Many thanks!
